static/app/components/feedback/feedbackItem/feedbackActions.tsx

```diff
@@ -29,73 +26,139 @@ export default function FeedbackActions({
+      {size === 'medium' ? <MediumWidth feedbackItem={feedbackItem} /> : null}
+      {size === 'small' ? <SmallWidth feedbackItem={feedbackItem} /> : null}
+    </Flex>
+  );
+}
+
+function LargeWidth({feedbackItem}: {feedbackItem: FeedbackIssue}) {
+  const {
+    isResolved,
+    onResolveClick,
+    hasSpamFeature,
+    isSpam,
+    onSpamClick,
+    hasSeen,
+    onMarkAsReadClick,
+  } = useFeedbackActions({feedbackItem});
+
+  return (
+    <Fragment>
+      <Button priority={isResolved ? 'danger' : 'primary'} onClick={onResolveClick}>
+        {isResolved ? t('Unresolve') : t('Resolve')}
       </Button>
+
       {hasSpamFeature && (
-        <Button
-          priority="default"
-          onClick={() => {
-            addLoadingMessage(t('Updating feedback...'));
-            const newStatus = isSpam ? GroupStatus.UNRESOLVED : GroupStatus.IGNORED;
-            resolve(newStatus, mutationOptions);
-            if (!isSpam) {
-              // not currently spam, clicking the button will turn it into spam
-              trackAnalytics('feedback.mark-spam-clicked', {
-                organization,
-                type: 'details',
-              });
-            }
-          }}
-        >
+        <Button priority="default" onClick={onSpamClick}>
           {isSpam ? t('Move to Inbox') : t('Mark as Spam')}
         </Button>
       )}
-      <Button
-        priority={isResolved ? 'danger' : 'primary'}
-        onClick={() => {
-          addLoadingMessage(t('Updating feedback...'));
-          const newStatus = isResolved ? GroupStatus.UNRESOLVED : GroupStatus.RESOLVED;
-          resolve(newStatus, mutationOptions);
-        }}
-      >
+
+      <Button onClick={onMarkAsReadClick}>
+        {hasSeen ? t('Mark Unread') : t('Mark Read')}
+      </Button>
+    </Fragment>
+  );
+}
+
+function MediumWidth({feedbackItem}: {feedbackItem: FeedbackIssue}) {
+  const {
+    isResolved,
+    onResolveClick,
+    hasSpamFeature,
+    isSpam,
+    onSpamClick,
+    hasSeen,
+    onMarkAsReadClick,
+  } = useFeedbackActions({feedbackItem});
+
+  return (
+    <Fragment>
+      <Button priority={isResolved ? 'danger' : 'primary'} onClick={onResolveClick}>
         {isResolved ? t('Unresolve') : t('Resolve')}
       </Button>
-    </Flex>
+
+      <DropdownMenu
+        position="bottom-end"
+        triggerProps={{
+          'aria-label': t('Action Menu'),
+          icon: <IconEllipsis />,
+          showChevron: false,
+          size: 'xs',
+        }}
+        items={[
+          hasSpamFeature
+            ? {
+                key: 'spam',
+                label: isSpam ? t('Move to Inbox') : t('Mark as Spam'),
+                onAction: onSpamClick,
+              }
+            : null,
+          {
+            key: 'read',
+            label: hasSeen ? t('Mark Unread') : t('Mark Read'),
+            onAction: onMarkAsReadClick,
+          },
+        ].filter(defined)}
```

Comment:
what does `defined` do here?

Reply:
`defined` is like doing `.filter(x => Boolean(x))` to verify that something is not-null, and it has the extra benefit that it's typed properly so TS knows that the result is not null or undefined. It's needed to make `items={[...]}` happy.